src/methods/wallet-get-calls-status.ts

```diff
@@ -1,85 +1,69 @@
 import { rpcErrors } from '@metamask/rpc-errors';
 import type { Infer } from '@metamask/superstruct';
-import {
-  nonempty,
-  optional,
-  mask,
-  string,
-  array,
-  object,
-  tuple,
-} from '@metamask/superstruct';
+import { tuple } from '@metamask/superstruct';
 import type {
+  Hex,
   Json,
   JsonRpcRequest,
   PendingJsonRpcResponse,
 } from '@metamask/utils';
-import { HexChecksumAddressStruct, StrictHexStruct } from '@metamask/utils';
+import { StrictHexStruct } from '@metamask/utils';
 
 import { validateParams } from '../utils/validation';
 
-const GetCallsStatusStruct = tuple([nonempty(string())]);
+const GetCallsStatusStruct = tuple([StrictHexStruct]);
 
-const GetCallsStatusReceiptStruct = object({
-  logs: optional(
-    array(
-      object({
-        address: optional(HexChecksumAddressStruct),
-        data: optional(StrictHexStruct),
-        topics: optional(array(StrictHexStruct)),
-      }),
-    ),
-  ),
-  status: optional(StrictHexStruct),
-  chainId: optional(StrictHexStruct),
-  blockHash: optional(StrictHexStruct),
-  blockNumber: optional(StrictHexStruct),
-  gasUsed: optional(StrictHexStruct),
-  transactionHash: optional(StrictHexStruct),
-});
+export enum GetCallsStatusCode {
+  PENDING = 100,
+  CONFIRMED = 200,
+  FAILED_OFFCHAIN = 400,
+  REVERTED = 500,
+  REVERTED_PARTIAL = 600,
+}
 
 export type GetCallsStatusParams = Infer<typeof GetCallsStatusStruct>;
-export type GetCallsStatusReceipt = Infer<typeof GetCallsStatusReceiptStruct>;
 
 export type GetCallsStatusResult = {
-  status: 'PENDING' | 'CONFIRMED';
-  receipts?: GetCallsStatusReceipt[];
+  version: string;
+  id: Hex;
+  chainId: Hex;
+  status: number;
+  receipts?: {
+    logs: {
+      address: Hex;
+      data: Hex;
+      topics: Hex[];
+    }[];
+    status: Hex;
+    blockHash: Hex;
+    blockNumber: Hex;
+    gasUsed: Hex;
+    transactionHash: Hex;
+  }[];
+  capabilities?: Record<string, Json>;
 };
 
-export type GetTransactionReceiptsByBatchIdHook = (
-  batchId: string,
+export type GetCallsStatusHook = (
+  id: GetCallsStatusParams[0],
   req: JsonRpcRequest,
-) => Promise<GetCallsStatusReceipt[]>;
+) => Promise<GetCallsStatusResult[]>;
```

Comment:
Should this return an array? I'm not seeing that in the spec, unless I missed it.

Reply:
Good spot, fixed.